### PR TITLE
feat(manifests): allows configuring manifest repositories using flags

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

As we are working on adding Service Mesh bits to the new operator I enhanced the manifests script so now one can call it using flags being named after components and pass alternative location. Please note it now also has repo org/user as first part.

```
get_all_manifests.sh --odh-dashboard="cam-garrison:odh-dashboard:test-manifests:manifests:odh-dashboard"
```

This will also work when building an image through build-args:

```
make image IMAGE_BUILD_FLAGS=" --build-arg USE_LOCAL=false --build-arg OVERWRITE_MANIFESTS='--odh-dashboard="cam-garrison:odh-dashboard:test-manifests:manifests:odh-dashboard"'"
```

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
would like to have updates in https://github.com/maistra/opendatahub-operator/blob/get_manifests_config/README.md#download-manifests, any maybe add the example usage as well

**Comment by bartoszmajsak** [comment]
> would like to have updates in [maistra/opendatahub-operator@`get_manifests_config`/README.md#download-manifests](https://github.com/maistra/opendatahub-operator/blob/get_manifests_config/README.md?rgh-link-date=2023-10-03T06%3A33%3A46Z#download-manifests), any maybe add the example usage as well

Thanks for this suggestion @zdtsw. In 7868e39 I extended the docs. In addition, commit 8678660 enhances the way how flags are handled, so in case if you pass non existing `--component=repo-org:repo-name:branch-name:source-folder:target-folder` script will add it to the list. Might be handy in testing stuff.

File: get_all_manifests.sh [hunk 1]
```diff
@@ -32,16 +59,20 @@ cp -r ./.odh-manifests-tmp/kserve/ ./odh-manifests
 ls -lat ./odh-manifests/
```
Comment (zdtsw) [inline get_all_manifests.sh:59]: wanna remove this one? thought i cleaned it up, but looks like it is still there :/

Comment (bartoszmajsak) [reply get_all_manifests.sh:56]:
```suggestion
```

**Pull request merged**